src/core_plugins/metrics/server/lib/get_index_pattern_service.js

```diff
@@ -0,0 +1,11 @@
+import { IndexPatternsService } from '../../../../server/index_patterns/service';
+export const getIndexPatternService = {
+  assign: 'indexPatternsService',
+  method(req, reply) {
+    const dataCluster = req.server.plugins.elasticsearch.getCluster('data');
```

Comment:
When https://github.com/elastic/kibana/pull/13012 gets merged this can be simplified to `req.getIndexPatternsService()`, and could be moved to into the routes if desired.